test/tests/pbs_admin_suspend.py

```diff
@@ -830,3 +830,35 @@ def test_hook(self):
         rc = self.mom.log_match("list of maintenance_jobs are %s" % (jid2),
                                 max_attempts=5)
         self.assertTrue(rc)
+
+    def test_offline(self):
+        """
+        Test that if a node is put to offline
+        and removed from maintenance state it
+        remains offlined
+        """
+
+        # submit a job and admin-suspend it
+        j = Job(TEST_USER)
+        jid1 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid1)
+        self.server.sigjob(jid1, 'admin-suspend')
+
+        # node state is in maintenance
+        self.server.expect(NODE, {'state': 'maintenance'}, id='vn[0]')
+
+        # submit another job. it will be queued
+        jid2 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
+
+        # mark the node as offline too
+        self.server.manager(MGR_CMD_SET, NODE, {'state': 'offline'},
+                            id='vn[0]')
+
+        # delete job1 as user
+        self.server.deljob(jid1, wait=True, runas=TEST_USER)
```

Comment:
I'd like to see this test modified slightly.  Instead of deleting the suspended job, I'd rather you admin-resume it.  The job will resume.  This will test the part of the feature where admin-resume directly resumes the job rather than waiting for the scheduler to resume it.